src/Composer/ComposerOrchestrator.php

```diff
@@ -39,12 +40,36 @@ final class ComposerOrchestrator
 {
     use NotInstantiable;
 
-    public static function getVersion(?string $composerBin = null): string
-    {
+    private const SUPPORTED_VERSION_CONSTRAINTS = '^2.2.0';
+
+    /**
+     * @throws IncompatibleComposerVersion
+     */
+    public static function checkVersion(
+        ?string $composerBin = null,
+        ?IO $io = null,
+    ): void {
+        $version = self::getVersion($composerBin, $io);
```

Comment:
I think it would make sense to cache the bin resolution; also maybe the bin & io could be passed in the constructor, i.e. making this service non-static